sherpa/astro/xspec/__init__.py

```diff
@@ -953,7 +953,7 @@ def calc(self, p, *args, **kwargs):
         #
         
         assert all((param.min <= value <= param.max 
-            for value, param in zip(self.pars, p))), "model parameters outside bounds"
+            for param, value in zip(self.pars, p))), "model parameters outside bounds"
```

Comment:
I think this would be better as (untested), since it matches the "normal" parameter error for an out-of-bands setting. I have also chose to use the hard bounds here (in XSPEC table-model terms, the `MINIMUM` and `MAXIMUM` values rather than the `TOP` and `BOTTOM` values), since this is a low-level function.

```
for param, value in zip(self.pars, p):
    if value < param._hard_min:
        raise ParameterErr('edge', self.name, 'minimum', param._hard_min)
    if val > param._hard_max:
        raise ParameterErr('edge', self.name, 'maximum', param._hard_max)
````